### PR TITLE
iterator with order

### PR DESCRIPTION
**Pull request by alwayslove2013**
issue: #758

### CODE REVIEW

File: include/knowhere/index/index_node.h [hunk 1]
```diff
@@ -243,6 +246,16 @@ class IndexIterator : public IndexNode::iterator {
         auto ret = q.top();
         q.pop();
         UpdateNext();
+        if (with_order_) {
+            while (HasNext()) {
+                auto next_ret = q.top();
+                if (next_ret.val >= ret.val) {
```
Comment (PwzXxm) [inline include/knowhere/index/index_node.h:252]: What about other metrics? Cosine/IP?

File: include/knowhere/config.h [hunk 2]
```diff
@@ -636,6 +636,7 @@ class BaseConfig : public Config {
     CFG_INT range_search_k;
     CFG_FLOAT range_filter;
     CFG_FLOAT range_search_level;
+    CFG_BOOL iterator_with_order;
```
Comment (zhengbuqian) [inline include/knowhere/config.h:639]: retain_iterator_order

File: include/knowhere/index/index_node.h (hunk 1 above)
Comment (PwzXxm) [reply include/knowhere/index/index_node.h:256]: Now Explained in the comments.

**Comment by alwayslove2013** [comment]
issue: #758

**Pull request merged**